<?xml version="1.0"?>
<launch>
    <arg name="output_type" default="screen"/>
    <arg name="bringup_self_manager" default="true"/>       <!--if you want to use other manager please set it as false-->
    <arg name="manager_name" default="keli_laser_manager"/> <!--if you want to use other manager please set it as the name what you specify-->
    <arg name="manager_name2" default="keli_laser_manager2"/>
    <param name="robot_description" command="$(find xacro)/xacro.py '$(find sdkeli_ls_udp)/urdf/sdkeli_ls1207de.urdf.xacro'" />
    <param name="robot_description2" command="$(find xacro)/xacro.py '$(find sdkeli_ls_udp)/urdf/sdkeli_ls1207de_2.urdf.xacro'" />
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />

    <group if="$(arg bringup_self_manager)">
        <node pkg="nodelet" type="nodelet" name="$(arg manager_name)" args="manager" output="screen"  respawn="false">
            <param name="num_worker_threads" value="12" />
        </node>
   	<node pkg="nodelet" type="nodelet" name="$(arg manager_name2)" args="manager" output="screen"  respawn="false">
            <param name="num_worker_threads" value="12" />
        </node>
    </group>

    <!--node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="1 0 0 0 0 0 1 laser laser2 50" /-->

	<!-- front laser -->
	<node pkg="tf" type="static_transform_publisher"
	   name="link1_broadcaster"
	   args="1 0 0 0 0 0 base_link laser 50" />

	<!-- back laser -->
	<node pkg="tf" type="static_transform_publisher"
	   name="link2_broadcaster"
	   args="1 0 0 0 0 0 base_link laser2 50" /> 

    <node pkg="nodelet" type="nodelet" name="sdkeli_ls1207de_nodelet" args="load sdkeli_ls_udp/SDKeliLs1207DENodelet $(arg manager_name)" output="$(arg output_type)"  respawn="false">
        <!--
          <param name="min_ang" type="double" value="-2.35619449019" />
          <param name="max_ang" type="double" value="2.35619449019" />
          <param name="intensity" type="bool" value="True" />
          <param name="skip" type="int" value="0" />
          <param name="frame_id" type="str" value="laser" />
          <param name="time_offset" type="double" value="-0.001" />
          <param name="publish_datagram" type="bool" value="False" />
          <param name="subscribe_datagram" type="bool" value="false" />
          <param name="device_number" type="int" value="0" />
          <param name="time_increment" type="double" value="0.000061722" />
          <param name="range_min" type="double" value="0.05" />
          <param name="range_max" type="double" value="10.0" />
        -->
        <remap from="scan" to="keli_scan" />	
        <param name="frame_id" type="str" value="laser" />
        <param name="range_min" type="double" value="0.01" />
        <param name="range_max" type="double" value="30.0" />
        <param name="hostname" type="string" value="192.168.0.10" />
        <param name="port" type="string" value="2112" />
        <param name="timelimit" type="int" value="5" />
        <param name="checkframe" type="bool" value="true" />


    </node>

    <node pkg="nodelet" type="nodelet" name="sdkeli_ls1207de_nodelet2" args="load sdkeli_ls_udp/SDKeliLs1207DENodelet $(arg manager_name2)" output="$(arg output_type)"  respawn="false">
        <!--
          <param name="min_ang" type="double" value="-2.35619449019" />
          <param name="max_ang" type="double" value="2.35619449019" />
          <param name="intensity" type="bool" value="True" />
          <param name="skip" type="int" value="0" />
          <param name="frame_id" type="str" value="laser" />
          <param name="time_offset" type="double" value="-0.001" />
          <param name="publish_datagram" type="bool" value="False" />
          <param name="subscribe_datagram" type="bool" value="false" />
          <param name="device_number" type="int" value="0" />
          <param name="time_increment" type="double" value="0.000061722" />
          <param name="range_min" type="double" value="0.05" />
          <param name="range_max" type="double" value="10.0" />
        -->
        <remap from="scan" to="keli_scan2" />
        <param name="frame_id" type="str" value="laser2" />
        <param name="range_min" type="double" value="0.01" />
        <param name="range_max" type="double" value="30.0" />
        <param name="hostname" type="string" value="192.168.0.11" />
        <param name="port" type="string" value="2112" />
        <param name="timelimit" type="int" value="5" />
        <param name="checkframe" type="bool" value="true" />


    </node>

</launch>

